src/platforms/generic_unix/lib/sys.c

```diff
@@ -559,6 +560,9 @@ void sys_init_platform(GlobalContext *global)
 #endif
 #endif
 
+    // TODO implement these functions in shlibs and load dynamically
+    // once we have a framework for that
```

Comment:
what worries you here?

Reply:
Having this static linkage in UNIX is okay, but I agree with your previous reviews that it would be good if we could dynamically load Nifs and Ports on platforms that support dlopen/dlsym

Reply:
It is not very documented, but we have it for port drivers.
See also src/platforms/generic_unix/lib/sys.c:488
We need it also for NIFs indeed anyway.
However for the sake of simplicity I would keep DNS resloving as part of the main executable, so I would remove this specific comment and I rather open an issue about the general topic.

Reply:
Removed comment